apps/mobile-wallet/src/features/auto-lock/useAutoLock.tsx

```diff
@@ -9,6 +9,7 @@ let lockTimer: number | undefined
 
 const useAutoLock = (unlockApp: () => Promise<void>) => {
   const appState = useRef<AppStateStatus>('active')
+  const unlockAppRef = useRef(unlockApp)
```

Comment:
Why was this change needed?

Reply:
To avoid that `unlockApp` identity (ref) changes trigger the useEffect below. I remember seeing this to be advised by the react team after React 19 (related to compiler optimisations): using refs prevents useless rerenders. I'll look for sources and docs.